core/corehttp/gateway_handler.go

```diff
@@ -162,26 +161,18 @@ func (i *gatewayHandler) getOrHeadHandler(ctx context.Context, w http.ResponseWr
 	dir := false
 	switch err {
 	case nil:
-		// core.Resolve worked
+		// Cat() worked
 		defer dr.Close()
 	case coreiface.ErrIsDir:
 		dir = true
-	case namesys.ErrResolveFailed:
-		// Don't log that error as it is just noise
-		w.WriteHeader(http.StatusInternalServerError)
-		fmt.Fprintf(w, "Path Resolve error: %s", err.Error())
-		log.Info("Path Resolve error: %s", err.Error())
-		return
 	case coreiface.ErrOffline:
 		if !i.node.OnlineMode() {
-			w.WriteHeader(http.StatusServiceUnavailable)
-			fmt.Fprint(w, "Could not resolve path. Node is in offline mode.")
+			webError(w, "ipfs cat "+urlPath, err, http.StatusServiceUnavailable)
 			return
 		}
 		fallthrough
 	default:
-		// all other erros
-		webError(w, "Path Resolve error", err, http.StatusBadRequest)
+		webError(w, "ipfs cat "+urlPath, err, http.StatusNotFound)
```

Comment:
The printed message contains `err`, so it's gonna be something like: "ipfs cat /ipns/foo.com: Could not resolve name"